lib/propolis/Cargo.toml

```diff
@@ -54,4 +54,4 @@ rand.workspace = true
 default = []
 crucible-full = ["crucible", "crucible-client-types", "oximeter", "nexus-client"]
 falcon = ["libloading", "p9ds", "dlpi", "ispf", "rand", "softnpu-lib", "num_enum"]
-omicron-build = []
+omicron-build = ["crucible/notify-nexus"]
```

Comment:
We should be requiring this feature directly via the propolis-server deps, not through `omicron-build` here in propolis-lib.

Reply:
I first tried

```patch
diff --git a/bin/propolis-server/Cargo.toml b/bin/propolis-server/Cargo.toml
index 6e30f39..4b71754 100644
--- a/bin/propolis-server/Cargo.toml
+++ b/bin/propolis-server/Cargo.toml
@@ -77,7 +77,7 @@ default = []
 
 # When building to be packaged for inclusion in the production ramdisk
 # (nominally an Omicron package), certain code is compiled in or out.
-omicron-build = ["propolis/omicron-build"]
+omicron-build = ["propolis/omicron-build", "crucible/notify-nexus"]
 
 # Falcon builds require corresponding bits turned on in the dependency libs
 falcon = ["propolis/falcon", "propolis_api_types/falcon"]
 ```
 
 but saw
 
 ```
 cargo build
error: failed to load manifest for workspace member `/home/james/src/oxidecomputer/propolis/bin/propolis-server`

Caused by:
  failed to parse manifest at `/home/james/src/oxidecomputer/propolis/bin/propolis-server/Cargo.toml`

Caused by:
  feature `omicron-build` includes `crucible/notify-nexus`, but `crucible` is not a dependency
 ```

Reply:
I think this would be more appropriate:
```diff
 default = []
-crucible-full = ["crucible", "crucible-client-types", "oximeter", "nexus-client"]
+crucible-full = ["crucible", "crucible/notify-nexus", "crucible-client-types", "oximeter", "nexus-client"]
 falcon = ["libloading", "p9ds", "dlpi", "ispf", "rand", "softnpu-lib", "num_enum"]
-omicron-build = ["crucible/notify-nexus"]
+omicron-build = []
```

We're expecting to have the nexus-involved behavior whenever `crucible-full` is set.  Why would we only build it in when `omicron-build` is set?

Reply:
I was only trying to avoid cases where an Upstairs would try to notify Nexus when in CI or when building propolis-server to run on its own (not in a full setup). Is that the case with `crucible-full` being set? Sorry, I get the intention of the `omicron-build` feature but not the `crucible-full` one

Reply:
Following up with stuff from the chat: I pushed eaa60e4 which adds a note about oxidecomputer/crucible#1280, which tracks the work to configure Nexus notifications at runtime instead of compile time.

Reply:
Thanks!